src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1207,25 +1207,20 @@ class SQLTargetWHandle : public StorageTargetWHandle {
         sha512Hash = hash.HashString();
       }
     }
-
-    auto statement = db_.prepareStatement<std::string, SQLZeroBlob>(
-        "INSERT OR REPLACE INTO target_images_data (filename, image_data) VALUES (?,?);", target_.filename(),
-        SQLZeroBlob{target_.length()});
-
-    if (statement.step() != SQLITE_DONE) {
-      LOG_ERROR << "Statement step failure: " << db_.errmsg();
-      throw exc;
-    }
-
-    row_id_ = sqlite3_last_insert_rowid(db_.get());
-    statement = db_.prepareStatement<std::string, std::string, std::string>(
+    auto statement = db_.prepareStatement<std::string, std::string, std::string>(
```

Comment:
If we handle the absence of a binary file correctly, we also can get rid of `db_.beginTransaction()` and `db_.commitTransaction()` here. This statement can act as an implicit transaction if I understand sqlite correctly (just how it works in `commit()`).